### ISSUE TITLE
deploy: implement GHA with replacing Cloud Build

### ISSUE THREAD

**Issue by hwakabh** [hwakabh/6ow3idGirl#41]
## AsIs
Currently we are using Google Cloud build for deploying application, but it is deactivated on the platform side.
Since this application will run on GKE, we can simply use GitHub Actions and build the container image for serve from GitHub packages.

## ToDo
- [x] replace Cloud Build with GitHub Actions
- [x] implement pipeline to push image to GitHub Packages

**Comment by hwakabh** [comment]
Confirmed that container image build by build packs has been uploaded to [GHCR](https://github.com/hwakabh/6ow3idGirl/pkgs/container/6ow3idgirl) as GitHub Packages.

As fixed in 3908837, note that we could not include upper case in tag name with GHCR. Otherwise, the process will be exited error with following errors.
```log
ERROR: failed to build: invalid image name 'ghcr.io/hwakabh/6ow3idGirl:latest': 'ghcr.io/hwakabh/6ow3idGirl:latest' is not a valid tag reference: could not parse reference: ghcr.io/hwakabh/6ow3idGirl:latest
Error: Process completed with exit code 1.
```

**Comment by hwakabh** [comment]
Confirmed that all workload has been deployed successfully with build image by GitHub Actions + Buildpacks, and we can see page rendered with temporary EXTERNAL-IP of Kubernetes service resource.
Note that we had manually updated serviceType with `kubectl edit` after manifests applied, since we need to manage external access with Kubernetes ingress resources instead of service whose type is LoadBalancer.
Also, the strange thing is we can not reach to express app within container with default port `tcp/3000`, so we have updated it with `tcp/8080` and mapped it with `tcp/80` for external access.

```shell
% kubectl -n 6ow3idgirl get all
NAME                           READY   STATUS    RESTARTS   AGE
pod/express-555d69797d-btk2d   1/1     Running   0          4m22s
pod/express-555d69797d-mhcn7   1/1     Running   0          4m6s
pod/express-555d69797d-rkmnj   1/1     Running   0          4m17s

NAME              TYPE           CLUSTER-IP      EXTERNAL-IP     PORT(S)        AGE
service/express   LoadBalancer   10.100.15.230   34.41.178.118   80:32389/TCP   6h26m

NAME                      READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/express   3/3     3            3           6h26m

NAME                                 DESIRED   CURRENT   READY   AGE
replicaset.apps/express-555d69797d   3         3         3       4m23s
replicaset.apps/express-5b579977b9   0         0         0       35m

% curl -D - -o /dev/null -s -X GET http://34.41.178.118
HTTP/1.1 200 OK
X-Powered-By: Express
Content-Type: text/html; charset=utf-8
Content-Length: 17396
ETag: W/"43f4-rnvat1TLASIZ3pj7i3KfZCFOXcA"
Date: Fri, 01 Mar 2024 13:22:09 GMT
Connection: keep-alive
Keep-Alive: timeout=5
```

<img width="1680" alt="Screenshot 2024-03-01 at 22 24 14" src="https://github.com/hwakabh/6ow3idGirl/assets/25563897/91347112-2f03-417b-8749-9eaf4407a294">
